tools/wpt/virtualenv.py

```diff
@@ -100,9 +103,33 @@ def activate(self):
             # https://github.com/web-platform-tests/wpt/issues/27377
             # https://github.com/python/cpython/pull/9516
             os.environ.pop('__PYVENV_LAUNCHER__', None)
-        path = os.path.join(self.bin_path, "activate_this.py")
-        with open(path) as f:
-            exec(f.read(), {"__file__": path})
+
+        # Setup the path and site packages as if we'd launched with the virtualenv active
+        bin_dir = os.path.join(self.path, "bin")
+        os.environ["PATH"] = os.pathsep.join([bin_dir] + os.environ.get("PATH", "").split(os.pathsep))
+        os.environ["VIRTUAL_ENV"] = self.path
+
+        prev_length = len(sys.path)
+
+        schemes = set(sysconfig.get_scheme_names())
```

Comment:
Converting the returned tuple to a set seems like it cannot make anything faster since there's just a single lookup in the set, it's just more code:

```suggestion
        schemes = sysconfig.get_scheme_names()
```